File: ios/downloader/MaazterDownloader.m
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(MaazterDownloader, RCTEventEmitter)
RCT_EXTERN_METHOD(add: (nonnull NSString *)contentUri
        encKey: (NSString *)enckey
        quality: (nonnull NSString *)quality
        data: (NSString *)data
        resolve: (RCTPromiseResolveBlock)resolve
        reject: (RCTPromiseRejectBlock)reject
)
RCT_EXTERN_METHOD(remove: (nonnull NSString *)contentId)
RCT_EXTERN_METHOD(pause: (nonnull NSString *)contentId reason: (NSInteger *)reason)
RCT_EXTERN_METHOD(resume: (nonnull NSString *)contentId)
RCT_EXTERN_METHOD(pauseAll)
RCT_EXTERN_METHOD(resumeAll)
RCT_EXTERN_METHOD(getTracks: (nonnull NSString *)contentUri
        encKey: (NSString *)enckey
        resolve: (RCTPromiseResolveBlock)resolve
        reject: (RCTPromiseRejectBlock)reject
)
RCT_EXTERN_METHOD(listDownloads: (RCTPromiseResolveBlock)resolve reject: (RCTPromiseRejectBlock)reject)
@end

File: ios/downloader/MaazterDownloader.swift
//
//  MaazterDownloaderModule.swift
//  MaazterPlayer
//
//  Created by Samar Yalini on 09/08/21.
//  Copyright © 2021 Facebook. All rights reserved.
//

import Foundation

@objc(MaazterDownloader)
class MaazterDownloader: RCTEventEmitter {

    @objc
    override static func requiresMainQueueSetup() -> Bool {
        true
    }

    override func supportedEvents() -> [String]! {
        return ["onDownloadChanged", "onDownloadRemoved", "onIdle"]
    }

    @objc
    func add(_ contentUri: String, encKey: String?, quality: String, data: String?, resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
        reject("E_NOT_IMPLEMENTED", "Not Implemented", NSError(domain: "", code: 200, userInfo: nil))
    }

    @objc
    func remove(_ contentId: String) {

    }

    @objc
    func pause(_ contentId: String, reason: Int = 1) {

    }

    @objc
    func pauseAll() {

    }

    @objc
    func resume(_ contentId: String) {

    }

    @objc
    func resumeAll() {

    }

    @objc
    func getTracks(_ contentUri: String, encKey: String?, resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
        reject("E_NOT_IMPLEMENTED", "Not Implemented", NSError(domain: "", code: 200, userInfo: nil))
    }

    @objc
    func listDownloads(_ resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {

    }
}
